include/device.h

```diff
@@ -240,6 +240,17 @@ __syscall struct device *device_get_binding(const char *name);
  */
 size_t z_device_get_all_static(struct device **devices);
 
+/** @brief Determine whether a device has been successfully initialized.
+ *
+ * @param dev pointer to the device in question.
+ *
+ * @return true if and only if the device is available for use.
+ */
+static inline bool z_device_ok(const struct device *dev)
```

Comment:
I have a similar patch for device const-ify stuff (name is different and obviously it does not check on driver_api attribute, plus other details). Since that one is not very relevant for that PR, could your remove that commit?

Reply:
I wondered if that might conflict, but figured that outside the potential name difference it should be easy to adapt to either.  Removed anyway.

I do think we should be consistent in naming, i.e. agree whether the check for validity  and fetch of static device pointers should be `device_*` or `z_device_*` functions.

Reply:
Can't close #24750 without that patch, though.

Reply:
indeed, I can pull my patch out for it then

Reply:
Hum it's inside a patch that revise the initialization status part. 

Well, let's cut the pear in half: call it device_ready() and that will be fine for me then

Reply:
I'll do that.

But if this is `device_ready` then perhaps `z_device_get_all_static()` should be `device_get_all_static()`.  I don't see applications having a use for `device_ready()`, as they can't get a device pointer unless it's ready.  (In a future Zephyr when they can, e.g. when devices can be made unready, we might have to add more API.)

Reply:
It's back, but I kept the internal-use prefix (which I may be mis-using, not sure).  This path allows `device_ready()` to remain available for future situations where somebody might get a device pointer without going through a validating API like `device_get_binding()`, or where a device may be made unready through power management.